frost-core/src/frost/keys.rs

```diff
@@ -565,12 +494,23 @@ pub(crate) fn generate_secret_shares<C: Ciphersuite>(
     Ok(secret_shares)
 }
 
-/// Recompute the secret from t-of-n secret shares using Lagrange interpolation.
-pub fn reconstruct_secret<C: Ciphersuite>(
+/// Recompute the secret from at least `min_signers` secret shares
+/// using Lagrange interpolation.
+///
+/// This can be used if for some reason the original key must be restored; e.g.
+/// if threshold signing is not required anymore.
+///
+/// This is NOT required to sign with FROST; the whole point of FROST is being
+/// able to generate signatures only using the shares, without having to
+/// reconstruct the original key.
+///
+/// The caller is responsible for providing at least `min_signers` shares;
+/// if less than that is provided, a different key will be returned.
+pub fn reconstruct_key<C: Ciphersuite>(
```

Comment:
`reconstruct_signing_key`? or maybe even `reconstruct()`?

Reply:
Good point. My reasoning for this name was based on the existing `keygen_with_dealer` function which also has "key" in the name. If we follow your suggestion, should we also go for `gen_with_dealer` or `generate_with_dealer`?

Reply:
I went with @dconnolly 's suggestion together with mine in 5346ec2. My rationale was the Rust guideline https://rust-lang.github.io/rfcs/0356-no-module-prefixes.html. I slightly prefer the longer version like @natalieesk but it seems more idiomatic this way